lua/graph.lua

```diff
@@ -1417,11 +1417,23 @@ function filter_tags_generic(kv)
 
   end
 
+  -- hov restrictions
   if ((kv["hov"] and kv["hov"] ~= "no") or kv["hov:lanes"] or kv["hov:minimum"]) then
 
     kv["hov_tag"] = "true"
 
-    if (kv["hov"] == "designated") then
+    local only_hov_allowed = kv["hov"] == "designated"
+    if only_hov_allowed then
+      if kv["hov:lanes"] then
+        for lane in (kv["hov:lanes"] .. '|'):gmatch("([^|]*)|") do
+          if lane and lane ~= "designated" then
```

Comment:
FYI, for the purpose of routing, `yes` and `designated` are more or less equivalent within access tags. So as far as I can tell, `yes` should be treated similarly to `designated`, and `designated` shouldn’t by itself prohibit any mode of transportation in particular.

/ref https://github.com/valhalla/valhalla/issues/1297#issuecomment-391767406